### DISCUSSION TITLE
Could JSONResponse provide JsonEncoder class option on JSONResponse.__init__ () ?

### DISCUSSION THREAD

**Discussion by lishukan** [encode/starlette#1281] [Ideas] [answered]
When I use the class -- to serialize(jsonify) a response, I find that I can not assign the special JsonEncoder . Sometime i have to rewrite the JsonResponse class ,like following code
`class JSONResponse(Response):
media_type = "application/json"
```
def __init__(self,*args,**kwargs):
    self.encoder = kwargs.get('encoder',None)
    super().__init__(*args)

def render(self, content: typing.Any) -> bytes:
    return json.dumps(
        content,
        ensure_ascii=False,
        allow_nan=False,
        indent=None,
        separators=(",", ":"),
        cls=self.encoder,
    ).encode("utf-8")`
```

i think it's necessary that provide custom jsonencoder  param for  the class JSONResponse.
 here my solution :  https://github.com/encode/starlette/pull/1251

**Comment by Kludex** [comment, marked as answer]
https://github.com/encode/starlette/issues/1250#issuecomment-993882247
